presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -1507,6 +1508,67 @@ private static void validateSphericalType(String function, OGCGeometry geometry,
         }
     }
 
+    @SqlNullable
+    @Description("Returns the area of a geometry on the Earth's surface")
+    @ScalarFunction("ST_Area")
+    @SqlType(DOUBLE)
+    public static Double stSphericalArea(@SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        if (geometry.isEmpty()) {
+            return null;
+        }
+
+        validateSphericalType("ST_Area", geometry, EnumSet.of(POLYGON));
```

Comment:
Could you add support for multi-polygons as well?

Reply:
Sure, but assuming they do not overlap ? ie simply loop through and sum all areas ?

Reply:
Yes, that's a safe assumption. All geospatial functions assume that input geometries are simple and valid, which, for multi-polygons means that they consist of a collection of non-overlapping polygons. A simple sum should work just fine.